### PR TITLE
readme file was written

### PR DESCRIPTION
**Pull request by QQBratka**

### CODE REVIEW

**Review by issugy — CHANGES_REQUESTED** [review]
-` This is a web application that show simulation...` --> `This is a web application that show a simulation...`
 - `REST naming convention` --> `RESTfull application`
 - tell smth about what users can do with ur application
 
 common mistakes in student's projects, check ur, please:
 - AppConfig class: line 48, setPackagesToScan("cinema.model")
 - WbConfig class: line 10, set basePackages = {"cinema.controller"}
 - MovieSession class: add names for columns
 - Order class: add join column annotation and name for column order_time
 - Role class: rename field roleName to name
 - RoleEnum class: rename to RoleName
 - RoleName class: move in the Role class
 - ShoppingCart class: add join table annotation
 - Ticket class: add name for column

**Review by Nick97-git — COMMENTED** [review]
README:
-  a simulation of a cinema user service (remove user)
- Implementation details in project (in -> of)
- Spring (Web, REST, Security) -> (Core, Web, Security)
- remove JPQL from technologies
- Depending on the role you can use different endpoints:
- fix the display of this part
Screenshot 2022-01-24 at 07 29 11
Also, remove words Both and Opportunities, it will be enough to write like this USER/ADMIN:
- fix this too
Screenshot 2022-01-24 at 07 37 01
- add /register endpoint to list of available endpoints
- Recommendations to run project: -> How to run the project:
- paraphrase the first three steps, so they sound like actions, also, add 6th step like launch the project
- paraphrase paragraph about created users and their passwords to make it more simple

File: .gitignore [hunk 1]
```diff
@@ -1,3 +1,3 @@
-.idea/*
-*.iml
+.idea/*
+*.iml
 target/*
```
Comment (issugy) [inline .gitignore]: empty line )

**Review by issugy — CHANGES_REQUESTED** [review]
common mistakes in student's projects, check ur, please:
  
 - AppConfig class: setPackagesToScan("cinema.model")
 - WbConfig class: set basePackages = {"cinema.controller"}
 - MovieSession class: add names for columns
 - Order class: add join column annotation and name for column order_time
 - Role class: rename field roleName to name
 - RoleEnum class: rename to RoleName
 - RoleName class: move in the Role class
 - ShoppingCart class: add join table annotation
 - Ticket class: add name for column

File: src/main/java/cinema/config/AppConfig.java [hunk 2]
```diff
@@ -14,7 +14,7 @@
 
 @Configuration
 @PropertySource("classpath:db.properties")
-@ComponentScan(basePackages = {"cinema"})
+@ComponentScan(basePackages = {"cinema.model"})
```
Comment (KseniiaMakarova) [inline src/main/java/cinema/config/AppConfig.java]: тут должен быть указан пакет со всеми бинами, а сейчас тут пакет model
соответственно, spring не сможет создать бины
ты запускал приложение?

исправить на cinema.model предлагалось ниже в этом классе, где ты конфигурируешь ORM

File: src/main/java/cinema/model/MovieSession.java [hunk 3]
```diff
@@ -1,66 +1,66 @@
...
+public class MovieSession {
+    @Id
+    @GeneratedValue(strategy = GenerationType.IDENTITY)
+    private Long id;
+    @ManyToOne
+    private Movie movie;
+    @ManyToOne
+    @JoinColumn(name = "cinema_hall_id")
+    private CinemaHall cinemaHall;
+    @JoinColumn(name = "show_time")
```
Comment (KseniiaMakarova) [inline src/main/java/cinema/model/MovieSession.java]:
```suggestion
    @Column(name = "show_time")
```

File: src/main/java/cinema/model/Order.java [hunk 4]
```diff
@@ -1,71 +1,71 @@
...
+public class Order {
+    @Id
+    @GeneratedValue(strategy = GenerationType.IDENTITY)
+    private Long id;
+    @OneToMany
+    @JoinTable(name = "orders_tickets",
+            joinColumns = @JoinColumn(name = "order_id"),
+            inverseJoinColumns = @JoinColumn(name = "ticket_id"))
+    private List<Ticket> tickets;
+    @JoinColumn(name = "order_time")
```
Comment (KseniiaMakarova) [inline src/main/java/cinema/model/Order.java]:
```suggestion
    @Column(name = "order_time")
```

File: src/main/java/cinema/model/ShoppingCart.java [hunk 5]
```diff
@@ -1,59 +1,59 @@
...
+import javax.persistence.Table;
+
+@Entity
+@Table(name = "shopping_carts")
+public class ShoppingCart {
+    @Id
+    private Long id;
+    @OneToMany
+    @JoinTable(name = "shopping_karts_tickets",
+            joinColumns = @JoinColumn(name = "shopping_kart_id"),
```
Comment (KseniiaMakarova) [inline src/main/java/cinema/model/ShoppingCart.java]:
```suggestion
    @JoinTable(name = "shopping_carts_tickets",
            joinColumns = @JoinColumn(name = "shopping_cart_id"),

```

**Review by KseniiaMakarova — CHANGES_REQUESTED** [review]

**Review by KseniiaMakarova — APPROVED** [review]

**Pull request merged**